src/features2d/mser.rs

```diff
@@ -3,167 +3,157 @@ use core::*;
-pub struct MSER {
-    value: *mut CMSER,
+pub struct Mser {
+    value: *mut native::cvsys_MSER,
 }
 
-impl MSER {
-    /// Creates a new maximally stable extremal region extractor criteria.
-    pub fn new(
-        delta: c_int,
-        min_area: c_int,
-        max_area: c_int,
-        max_variation: f64,
-        min_diversity: f64,
-        max_evolution: c_int,
-        area_threshold: f64,
-        min_margin: f64,
-        edge_blur_size: c_int,
-    ) -> Self {
-        let mser = unsafe {
-            cv_mser_new(
-                delta,
-                min_area,
-                max_area,
-                max_variation,
-                min_diversity,
-                max_evolution,
-                area_threshold,
-                min_margin,
-                edge_blur_size,
-            )
-        };
-        MSER { value: mser }
-    }
-
+impl Mser {
     /// Detect MSER regions.
     pub fn detect_regions(&self, image: &Mat) -> (Vec<Vec<Point2i>>, Vec<Rect>) {
-        let mut msers = CVec::<CVec<Point2i>>::default();
-        let mut bboxes = CVec::<Rect>::default();
+        let mut msers: native::cvsys_CVec<native::cvsys_CVec<native::cvsys_Point2i>> = unsafe { std::mem::zeroed() };
+        let mut bboxes: native::cvsys_CVec<native::cvsys_Rect> = unsafe { std::mem::zeroed() };
         unsafe {
-            cv_mser_detect_regions(self.value, image.inner, &mut msers, &mut bboxes);
+            native::cvsys_mser_detect_regions(self.value, image.inner, &mut msers, &mut bboxes);
         }
-        let msers = msers.unpack();
-        let boxes = bboxes.unpack();
-        (msers, boxes)
+        let msers = msers
+            .iter()
+            .map(|inner| inner.iter().cloned().map(Into::into).collect())
+            .collect();
+        let bboxes = bboxes.iter().cloned().map(Into::into).collect();
+        (msers, bboxes)
     }
 }
 
-impl Drop for MSER {
+impl Drop for Mser {
     fn drop(&mut self) {
         unsafe {
-            cv_mser_drop(self.value);
+            native::cvsys_mser_drop(self.value);
         }
     }
 }
 
-/// Builder that provides defaults for MSER
-#[derive(Debug, Copy, Clone, Default)]
-pub struct MSERBuilder {
-    delta: Option<c_int>,
-    min_area: Option<c_int>,
-    max_area: Option<c_int>,
-    max_variation: Option<f64>,
-    min_diversity: Option<f64>,
-    max_evolution: Option<c_int>,
-    area_threshold: Option<f64>,
-    min_margin: Option<f64>,
-    edge_blur_size: Option<c_int>,
+impl Default for Mser {
+    fn default() -> Self {
+        MserBuilder::default().into()
+    }
 }
 
-impl MSERBuilder {
+/// Builder that provides defaults for [Mser](struct.Mser.html).
+///
+/// ## Defaults
+/// - delta: 5,
+/// - min_area: 60,
+/// - max_area: 14400,
+/// - max_variation: 0.25,
+/// - min_diversity: 0.2,
+/// - max_evolution: 200,
+/// - area_threshold: 1.01,
+/// - min_margin: 0.003,
+/// - edge_blur_size: 5,
+#[derive(Debug, Copy, Clone)]
+pub struct MserBuilder {
```

Comment:
It's not a builder since builder is working with optional fields. For example there is no way to determine if user uses `5` intentionally or it comes from default. And it could create a regression if/when opencv changes defaults and we do the same.

Reply:
The defaults are set in the `Default::default()` implementation. Rust has no default parameters or anything of the like. We could pass the options into C++, but since these are arguments (in C++), we must specify all of the arguments that come before one that we specify. This means that we must provide a default value and cannot rely on the ones from OpenCV. For this reason, all the values are filled and they are all passed every time on the C++ side.

If there is some advantage to changing the fields to be an `Option` (such as to optionally do something on the C++ side), please let me know and I can try to reconcile that situation.